### PR TITLE
hardened_unpack: Permit the snapshots/status_cache file to be sparse

### PR DESCRIPTION
**Pull request by mvines**
We had a report of a node that packaged `snapshots/status_cache` as sparse in its snapshot.   This is fine, allow it

### CODE REVIEW

File: runtime/src/hardened_unpack.rs [hunk 1]
```diff
@@ -206,6 +206,7 @@ fn is_valid_snapshot_archive_entry(parts: &[&str], kind: tar::EntryType) -> bool
         (["accounts", file], GNUSparse) if like_storage(file) => true,
         (["accounts", file], Regular) if like_storage(file) => true,
         (["snapshots"], Directory) => true,
+        (["snapshots", "status_cache"], GNUSparse) => true,
         (["snapshots", "status_cache"], Regular) => true,
         (["snapshots", dir, file], Regular) if all_digits(dir) && all_digits(file) => true,
```
Comment (t-nelson) [inline runtime/src/hardened_unpack.rs:211]: Should we go ahead and allow here as well?

Comment (mvines) [reply runtime/src/hardened_unpack.rs:211]: yep, might as well. done

**Comment by mvines** [comment]
```
failures:
--
  | test_associated_token_address
  | test_create_with_a_lamport
```

**Review by t-nelson — APPROVED** [review]
:shipit:

**Comment by mvines** [comment]
Green enough!

**Pull request merged**